File: 백준/Gold/11404. 플로이드/플로이드.py
import sys
import math
input = sys.stdin.readline

n=int(input()) #도시개수
m=int(input()) #버스개수



cost = [[math.inf for _ in range(n+1)] for _ in range(n+1)]
for i in range(1, n+1):
    cost[i][i] = 0
    
for _ in range(m):
    s, e, wt = map(int, input().split()) #시작, 도착, 비용
    if cost[s][e] > wt:
        cost[s][e] = wt
    
# for i in range(1, n+1):
#     for j in range(1, n+1):
#         print(cost[i][j], end = ' ')
#     print()

#플로이드
for k in range(1, n+1):
    for start in range(1, n+1):
        for end in range(1, n+1):
            # if start != end:
            if cost[start][end] > cost[start][k]+cost[k][end]:
                cost[start][end] = cost[start][k]+cost[k][end]

for i in range(1, n+1):
    for j in range(1, n+1):
        if cost[i][j] != math.inf:
            print(cost[i][j], end = ' ')
        else:
            print(0, end = ' ')
    print()

File: 백준/Gold/11404. 플로이드/플로이드.cc
#include <iostream>
#include <vector>
#include <utility>

using namespace std;

typedef pair<int, int> pii;
const int INF = 1e9; // 1,000,000,000

int main()
{

	int n, m; // 도시개수n, 버스개수m

	cin >> n >> m;

	vector<vector<int>> dist(n+1, vector<int>(n+1, INF));

	int start, end, cost;
	for (int idx = 0; idx < m; idx++)
	{
		cin >> start >> end >> cost;
		if (dist[start][end] > cost)
			dist[start][end] = cost;
	}

	for (int mid = 1; mid <= n; mid++)
	{
		for (int st = 1; st <= n; st++)
		{
			for (int end = 1; end <= n; end++)
			{
				
				if (dist[st][end] > dist[st][mid] + dist[mid][end])
					dist[st][end] = dist[st][mid] + dist[mid][end];
			}
		}
	}

	for (int st = 1; st <= n; st++)
	{
		for (int end = 1; end <= n; end++)
		{
			if (st== end || dist[st][end] == INF)
				cout << 0 << ' ';
			else
				cout << dist[st][end] << ' ';
		}
		cout << '\n';
	}
		
}